src/averageAges.js

```diff
@@ -55,7 +70,27 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const personNames = people.map(person => person.name);
+
+  const childrenSons = onlyWithSon
+    ? people.filter(person => person.mother !== null
+      && personNames.includes(person.mother)
+      && person.sex === 'm')
+    : people.filter(person => person.mother !== null
+      && personNames.includes(person.mother));
+
+  const namesOfMothers = childrenSons.map(person => person.mother);
+  const mothers = people.filter(person => namesOfMothers.includes(person.name));
```

Comment:
First map is redundant here
You can use some in filter method and check if some of people is mother of selected child